### PR TITLE
[SOC-2407] Create Gateway Docs for new Social Endpoints.

### PR DESCRIPTION
**Pull request by wfehr-va**
## [SOC-2407] Create Gateway Docs for new Social Endpoints.

Linked Confluence documents for context:
https://vendasta.jira.com/wiki/spaces/API/pages/1634369796/SM+Scheduling+Post+Design+Session
https://vendasta.jira.com/wiki/spaces/API/pages/1634009213/SM+-+Social+Profile+Endpoint+Design
https://vendasta.jira.com/wiki/spaces/API/pages/1633059029/SM+-+Schedule+Message+Endpoint+Design

This PR adds information for the two APIs we wish to implement in the API Gateway: Retrieving Social Profiles, and Scheduling Messages (Posts). It also adds entries for all the models that are used by them.

Link to versioned branch documentation: https://developers.vendasta.com/platform/branches/SOC-2407/ZG9jOjEwMTkzMDgz-change-log

@vendasta/tots

[SOC-2407]: https://vendasta.jira.com/browse/SOC-2407?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: openapi/social/social.yaml [hunk 1]
```diff
@@ -33,25 +35,37 @@ paths:
...
+            type: string
+            enum:
+              - ACTIVE
+              - AUTHORIZATION_REQUIRED
+              - DISABLED
           in: query
-          name: 'filter[canSchedule]'
-          description: 'Returns only the Social Profiles that can be used to schedule Messages. Defaults to `true` when using `filter[businessLocation]`.'
+          name: 'filter[status]'
+          description: Returns only the statuses that are defined.
```
Comment (richard-rance) [inline openapi/social/social.yaml:65]:
```suggestion
          description: Returns only the Social Profiles having one of the specified statuses. You may send multiple as a comma-separated list. 
```

File: openapi/social/social.yaml [hunk 2]
```diff
@@ -33,25 +35,37 @@ paths:
...
         - schema:
             type: string
           in: query
           name: 'filter[partnerIds]'
           description: 'Returns all Social Profiles under the given partnerId. Cannot be used with `filter[businessLocation]`.'
+          required: true
         - schema:
             type: string
+            enum:
+              - FACEBOOK
```
Comment (richard-rance) [inline openapi/social/social.yaml:49]: Please make the enum values camel case

File: openapi/social/social.yaml [hunk 3]
```diff
@@ -62,8 +76,11 @@ paths:
                 type: object
                 properties:
                   '':
-                    type: string
-      description: 'If this endpoint is given a message without an id, this endpoint will schedule it to be posted (this to defaults to posting in 5 minutes if no scheduledAt is given).'
+                    $ref: '#/components/schemas/messages'
+      description: |-
+        If this endpoint is given a message without an id, this endpoint will schedule it to be posted (this to defaults to posting in 5 minutes if no scheduledAt is given).
```
Comment (richard-rance) [inline openapi/social/social.yaml:81]:
```suggestion
        This operation will schedule a new message to be posted at the time defined in `sendAt`.  (this to defaults to posting in 5 minutes if no `sendAt` is given).
```

File: openapi/social/social.yaml (hunk 2 above)
Comment (wfehr-va) [reply openapi/social/social.yaml]: Done 👍

File: openapi/social/social.yaml [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+  - url: 'https://demo.apigateway.co/product/social'
+    description: Demo
+  - url: 'http://localhost:3000/product/social'
+    description: Local
+paths:
+  /socialProfiles:
+    get:
+      summary: List Social Profiles
+      tags:
+        - socialProfiles
```
Comment (richard-rance) [inline openapi/social/social.yaml:18]: These are displayed to users in the navigation.
```suggestion
        - "Social Profiles"
```

File: openapi/social/social.yaml [hunk 5]
```diff
@@ -0,0 +1,276 @@
+openapi: 3.1.0
+info:
+  title: Social
+  version: '1.0'
+  description: These APIs allow users to use Vendasta's social capabilities. All the endpoints require either the `social` or `business` scopes to be accessed.
+servers:
```
Comment (richard-rance) [inline openapi/social/social.yaml:6]: Please copy the security schema from the platform and add your oauth scopes

File: openapi/social/social.yaml [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+  - url: 'https://prod.apigateway.co/product/social'
```
Comment (richard-rance) [inline openapi/social/social.yaml:7]:
```suggestion
  - url: 'https://prod.apigateway.co/products/social'
```

File: openapi/social/social.yaml (hunk 5 above)
Comment (wfehr-va) [reply openapi/social/social.yaml:6]: Sounds good, done 👍

**Pull request merged**